pkg/kubelet/dockertools/manager.go

```diff
@@ -1856,24 +1868,26 @@ func (dm *DockerManager) syncPodWithSyncResult(pod *api.Pod, _ api.PodStatus, po
 
 		setupNetworkResult := kubecontainer.NewSyncResult(kubecontainer.SetupNetwork, kubecontainer.GetPodFullName(pod))
 		result.AddSyncResult(setupNetworkResult)
-		// Call the networking plugin
-		err = dm.networkPlugin.SetUpPod(pod.Namespace, pod.Name, podInfraContainerID)
-		if err != nil {
-			// TODO: (random-liu) There shouldn't be "Skipping pod" in sync result message
-			message := fmt.Sprintf("Failed to setup network for pod %q using network plugins %q: %v; Skipping pod", format.Pod(pod), dm.networkPlugin.Name(), err)
-			setupNetworkResult.Fail(kubecontainer.ErrSetupNetwork, message)
-			glog.Error(message)
-
-			// Delete infra container
-			killContainerResult := kubecontainer.NewSyncResult(kubecontainer.KillContainer, PodInfraContainerName)
-			result.AddSyncResult(killContainerResult)
-			if delErr := dm.KillContainerInPod(kubecontainer.ContainerID{
-				ID:   string(podInfraContainerID),
-				Type: "docker"}, nil, pod, message); delErr != nil {
-				killContainerResult.Fail(kubecontainer.ErrKillContainer, delErr.Error())
-				glog.Warningf("Clear infra container failed for pod %q: %v", format.Pod(pod), delErr)
+		if pod.Spec.SecurityContext != nil && !pod.Spec.SecurityContext.HostNetwork {
+			// Call the networking plugin
+			err = dm.networkPlugin.SetUpPod(pod.Namespace, pod.Name, podInfraContainerID)
+			if err != nil {
+				// TODO: (random-liu) There shouldn't be "Skipping pod" in sync result message
+				message := fmt.Sprintf("Failed to setup network for pod %q using network plugins %q: %v; Skipping pod", format.Pod(pod), dm.networkPlugin.Name(), err)
+				setupNetworkResult.Fail(kubecontainer.ErrSetupNetwork, message)
+				glog.Error(message)
+
+				// Delete infra container
+				killContainerResult := kubecontainer.NewSyncResult(kubecontainer.KillContainer, PodInfraContainerName)
+				result.AddSyncResult(killContainerResult)
+				if delErr := dm.KillContainerInPod(kubecontainer.ContainerID{
+					ID:   string(podInfraContainerID),
+					Type: "docker"}, nil, pod, message); delErr != nil {
+					killContainerResult.Fail(kubecontainer.ErrKillContainer, delErr.Error())
+					glog.Warningf("Clear infra container failed for pod %q: %v", format.Pod(pod), delErr)
+				}
+				return
 			}
-			return
 		}
 
 		// Setup the host interface unless the pod is on the host's network (FIXME: move to networkPlugin when ready)
```

Comment:
determineContainerIP() here cannot possibly return any interesting result here for two reasons:

1) if the network plugin wasn't called to set up the container, it seems really odd to ask the network plugin for the containers IP address.  Yes, this might work, but is really weird.

2) docker '--net=host' doesn't report any IP address for the container, so plugins that chain up to DockerManager.GetContainerIP() won't return anything either.

# sudo docker run --net=host -it dcbw/fedora-net /bin/bash
...
# sudo docker inspect 0a0685f13f6c | grep "NetworkMode\|IPAddress"
        "IPAddress": "",
        "SecondaryIPAddresses": null,
        "NetworkMode": "host",

while the patch LGTM now, I think there's definitely room for further optimization.  I'm happy to do that if you like.